### ISSUE TITLE
One of no-unnecessary-generics bad case is needed in some cases

### ISSUE THREAD

**Issue by shibukawa** [Microsoft/dtslint#212]
https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md

The following sample says the extends interface for input is bad design. 
----

**Bad**:

```ts
function useLogger<T extends Logger>(logger: T): void;
```

**Good**:

```ts
function useLogger(logger: Logger): void;
```

----

But some times, The interface includes extra data for other components. For example, command-line-args and command-line-usage shares almost all properties, but some properties are used for each package. 

In that case, I know only two solutions:

```
// 1: Add extra key-value fields
interface CommandOption {
   :
   [propName: string]: any;
}

// 2: Use extends and generics
function commandLineArgs<T extends CommandOption>(t: T);
```

I feel second one is better, because CommandOption for the function doesn't have any extra parameters. But I couldn't use the second option because of no-unnecessary-generics error.

thanks,
